concent_api/core/models.py

```diff
@@ -73,3 +75,216 @@ class MessageAuth(Model):
     message              = OneToOneField(StoredMessage, related_name = 'auth')
     provider_public_key  = Base64Field(max_length=64)
     requestor_public_key = Base64Field(max_length=64)
+
+
+class ClientManager(Manager):
+
+    def get_or_create_full_check(self, public_key):
+        """
+        Returns Model instance.
+        Does the same as get_or_create method, but also performs full_clean() on newly created instance.
+        """
+        try:
+            instance = self.get(public_key = public_key)
+        except self.model.DoesNotExist:
+            instance = self.model(
+                public_key = public_key
+            )
+            instance.full_clean()
+            instance.save()
+        return instance
+
+
+class Client(models.Model):
+    """
+    Represents Concent client, identified by public key.
+    """
+
+    objects = ClientManager()
+
+    public_key = Base64Field(max_length = 64, unique = True)
+
+
+class Subtask(models.Model):
+    """
+    Represents subtasks states.
+    """
+
+    class SubtaskState(ChoiceEnum):
+        ForcingReport           = 'forcing_report'
+        Reported                = 'reported'
+        ForcingResultTransfer   = 'forcing_result_transfer'
```

Comment:
Why camel case (`ForcingReport`) and not all caps (`FORCING_REPORT`)? Enum values are basically just constants.

Reply:
:+1: